File: pkg/codegen/testing/test/testdata/inline-invoke-pp/go/inline-invoke.go
package main

import (
	"github.com/pulumi/pulumi-aws/sdk/v4/go/aws"
	"github.com/pulumi/pulumi-aws/sdk/v4/go/aws/ec2"
	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
)

func main() {
	pulumi.Run(func(ctx *pulumi.Context) error {
		_, err := ec2.NewInstance(ctx, "webServer", &ec2.InstanceArgs{
			Ami: aws.GetAmiOutput(ctx, GetAmiResultOutputArgs{
				Filters: []map[string]interface{}{
					map[string]interface{}{
						"name": "name",
						"values": []string{
							"amzn-ami-hvm-*-x86_64-ebs",
						},
					},
				},
				Owners: []string{
					"137112412989",
				},
				MostRecent: true,
			}, nil).Id,
		})
		if err != nil {
			return err
		}
		return nil
	})
}

File: pkg/codegen/testing/test/testdata/inline-invoke-pp/nodejs/inline-invoke.ts
import * as pulumi from "@pulumi/pulumi";
import * as aws from "@pulumi/aws";

const webServer = new aws.ec2.Instance("webServer", {ami: aws.getAmi({
    filters: [{
        name: "name",
        values: ["amzn-ami-hvm-*-x86_64-ebs"],
    }],
    owners: ["137112412989"],
    mostRecent: true,
}).then(invoke => invoke.id)});
